clients/src/main/java/org/apache/kafka/common/config/ConfigDef.java

```diff
@@ -76,7 +76,7 @@
      * A unique Java object which represents the lack of a default value.<p>
      * The 'new' here is intentional.
      */
-    public static final Object NO_DEFAULT_VALUE = new String("");
+    public static final Object NO_DEFAULT_VALUE = new Object();
```

Comment:
We saw this issue in Confluent's copy of this stuff. An intended cleanup ended up breaking the code. I think by changing the type we also get rid of that findbugs warning (and other ones elsewhere in the file).

I think the problem is that the `==` *didn't* make it immediately clear, and we introduced a bug in a copy of this code that multiple people missed during review (including one, myself, that has had to deal with this class repeatedly).

In any case, sentinels that rely on uniqueness should be guaranteed to work with `==` and `equals` equality by using a type that ensures `equals` uses `==` under the hood, and a plain old `Object` accomplishes that.

Anyway, I don't feel that strongly about it, but as we're trying to get better tooling around the code, we'll have to deal with those findbugs warnings one way or another. But the Connect fix was truly a bug, so if we don't merge the ConfigDef changes, I'd at least like to fix the Connect part.